src/arcaflow_plugin_sdk/schema.py

```diff
@@ -4887,13 +4893,14 @@ class ObjectType(ObjectSchema, AbstractType, Generic[ObjectT]):
     You can now use the object_type to unserialize, validate, and serialize properties.
     """  # noqa: E501
 
-    _cls: Type[ObjectT]
-    properties: Dict[str, PropertyType]
+    _cls: Type[ObjectT] = None
+    properties: Dict[str, PropertyType] = None
+    loose_id: Optional[bool] = None
```

Comment:
This change is a bit weird. Or rather maybe it's normal in a weird context. Normally you don't define an `__init__` for an `@dataclass`, because you're essentially overriding the default init behavior that makes a dataclass a dataclass.

In the end, you're defining class level variables here (because of the values), which will only be used if code references an instance variable that doesn't have a key in the instance `__dict__` (providing a generic class-level default); and that'll never happen because your `__init__` always supplies instance-level values. (Actually, I think it's even weirder, and possibly a bit dangerous, to be calling the `ObjectSchema.__init__` here, since that's controlled by the `@dataclass` decorator, rather than a normal superclass constructor. It'll probably (and apparently does) work, but I think it's semantically weird ...

Reply:
It's a little odd. I agree. It has been this way since we took over the code from Janos. https://github.com/arcalot/arcaflow-plugin-sdk-python/blob/15426cd61dc7f035b36a331aab0f4fc0ec2b1624/src/arcaflow_plugin_sdk/schema.py#L4540